app/batid/views.py

```diff
@@ -35,3 +42,67 @@ def test_func(self):
     @classmethod
     def as_url(cls):
         return re_path(r"^(?P<path>{}.*)$".format(cls.url_prefix), cls.as_view())
+
+
+def contribution(request, contribution_id):
+    if not request.user.is_superuser:
+        return HttpResponseForbidden()
+    else:
+        contribution = Contribution.objects.get(id=contribution_id)
+        return render(
+            request,
+            "contribution.html",
+            {
+                "contribution_id": contribution_id,
+                "rnb_id": contribution.rnb_id,
+                "text": contribution.text,
+                "review_comment": contribution.review_comment,
+            },
+        )
+
+
+def delete_building(request):
+    if not request.user.is_superuser:
+        return HttpResponseForbidden()
+    else:
+        # check if the request is a POST request
+        if request.method == "POST":
+            # get the rnb_id from the request
+            rnb_id = request.POST.get("rnb_id")
+            contribution_id = request.POST.get("contribution_id")
+            review_comment = request.POST.get("review_comment")
+            contribution = Contribution.objects.get(id=contribution_id)
```

Comment:
Un appel `.get()` ne retournant pas de résultat fait crasher django.
On peut faire : 
```suggestion
            try:
                contribution = Contribution.objects.get(id=contribution_id)
            except Contribution.DoesNotExist:
                # Return a 404 ?
                raise HttpResponseNotFound(f"We could not find find contribution {contribution_id}")
```

Reply:
je découvre l'existence de cette fonction très pratique https://docs.djangoproject.com/fr/5.0/topics/http/shortcuts/#get-object-or-404